File: pinyougou_manager/pinyougou_manager_interface/src/main/java/com/pinyougou/manager/service/SpecificationService.java
package com.pinyougou.manager.service;

import com.pinyougou.common.PageResult;
import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.pojo.vo.SpecificationVo;

import java.util.List;
import java.util.Map;

public interface SpecificationService {

	/**
	 * 根据条件分页查询
	 * @param pageNum
	 * @param pageSize
	 * @param specification
	 * @return
	 */
	PageResult search(Integer pageNum, Integer pageSize, TbSpecification specification);

	/**
	 * 新增规格
	 * @param vo
	 */
	void add(SpecificationVo vo);

	/**
	 * 根据id回显实体类
	 * @param id
	 * @return
	 */
	public SpecificationVo findOne(long id);

	/**
	 * 修改
	 * @param vo
	 */
	public void update(SpecificationVo vo);

	/**
	 * 根据多个id删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 规格下拉框数据
	 * @return
	 */
	List<Map> findMap();
}

File: pinyougou_manager_web/src/main/webapp/js/service/specificationService.js
app.service("specificationService",function($http){
    this.search = function (pageNum, pageSize, entity) {
        return $http.post("../specification/search?pageNum="+pageNum+"&pageSize="+pageSize,entity);
    }
    this.add = function (entity) {
        return $http.post("../specification/add",entity);
    }
    this.findOne = function (id) {
        return $http.get("../specification/findOne?id="+id);
    }
    this.update = function (entity) {
        return $http.post("../specification/update",entity);
    }
    this.dele = function (ids) {
        return  $http.post("../specification/delete?ids="+ids);
    }

    //规格下拉列表数据
    this.findMap = function () {
        return $http.get("../specification/findMap");
    }
});